r/R/dplyr.R

```diff
@@ -74,7 +74,13 @@ print.arrow_dplyr_query <- function(x, ...) {
     name <- expr$field_name
     if (nzchar(name)) {
       # Just a field_ref, so look up in the schema
-      schm$GetFieldByName(name)$type$ToString()
+      field_name <- schm$GetFieldByName(name)
```

Comment:
I checked with @bkietz and (currently at least) evaluating expressions on a dataset with duplicated field names will error, so if that's the case (perhaps you could verify), we should check this sooner than the print method. How about instead we put a check around L33 above, like

```r
if (anyDuplicated(names(.data))) {
  abort("useful message")
}
```

Reply:
Have updated now

Reply:
Specifically: looking up fields by name when that name is not unique in the schema is not supported.